Mathlib/LinearAlgebra/TensorProduct.lean

```diff
@@ -216,12 +211,17 @@ theorem smul_tmul [DistribMulAction R' N] [CompatibleSMul R R' M N] (r : R') (m
   CompatibleSMul.smul_tmul _ _ _
 #align tensor_product.smul_tmul TensorProduct.smul_tmul
 
-attribute [local instance] addMonoid
+-- porting note: This is added as a local instance for `SMul.aux`.
+-- For some reason type-class inference in Lean 3 unfolded this definition.
+private def addMonoidWithWrongNSMul : AddMonoid (M ⊗[R] N) :=
+  { (addConGen (TensorProduct.Eqv R M N)).addMonoid with }
+
+attribute [local instance] addMonoidWithWrongNSMul
```

Comment:
```suggestion
attribute [local instance] addMonoidWithWrongNSMul in
```
Does it work so that you don't need to disable it on line 224?

Reply:
Yes, that works nicely; thanks!